jsaddle/src/Language/Javascript/JSaddle/Types.hs

```diff
@@ -875,27 +876,6 @@ instance MonadRef JSM where
 instance MonadAtomicRef JSM where
     atomicModifyRef r = liftIO . MonadRef.atomicModifyRef r
 
---TODO: Figure out what syncAfter was doing in MonadCatch and MonadMask, and do that
-instance MonadCatch JSM where
-    t `catch` c = JSM (unJSM t `catch` \e -> unJSM (c e))
-
-instance MonadMask JSM where
-  mask a = JSM $ mask $ \unmask -> unJSM (a $ q unmask)
-    where q :: (ReaderT JSContextRef IO a -> ReaderT JSContextRef IO a) -> JSM a -> JSM a
-          q unmask (JSM b) = JSM $ unmask b
-  uninterruptibleMask a =
-    JSM $ uninterruptibleMask $ \unmask -> unJSM (a $ q unmask)
-      where q :: (ReaderT JSContextRef IO a -> ReaderT JSContextRef IO a) -> JSM a -> JSM a
-            q unmask (JSM b) = JSM $ unmask b
-#if MIN_VERSION_exceptions(0,9,0)
```

Comment:
We should probably verify exceptions-0.9.0 still works. If not, just bump the bounds in jsaddle.cabal.

Reply:
Since we removed the custom instance, it should not cause any issue. The derived instance will be according to the version of exceptions.